.github/workflows/publish.yml

```diff
@@ -1,5 +1,8 @@
 name: Publish nightly snapshots and documentation
 
+permissions:
+  contents: read
```

Comment:
Afaiu, giving no permissions is equivalent to setting all permissions to `none`. I.e. we are giving more permissions here than before.